### ISSUE TITLE
bug: notifier: style minimal: can't change background

### ISSUE THREAD

**Issue by edshamis** [folke/snacks.nvim#410]
### Did you check docs and existing issues?

- [X] I have read all the snacks.nvim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of snacks.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.3

### Operating system/version

Ubuntu 22.04.5

### Describe the bug

With `notifier = {style = "minimal"}`
running `hi SnacksNotifierInfo guibg=black` has no effect:
![241230-112125](https://github.com/user-attachments/assets/fa3ddeea-1266-400c-bf62-09751f313493)

While with no style set it works:
![241230-112253](https://github.com/user-attachments/assets/012ae123-6c65-44cf-a9aa-609d1827f07d)

### Steps To Reproduce

1. run with style minimal
2. try to change background

### Expected Behavior

change background for minimal style as well

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "folke/snacks.nvim", opts = {} },
    -- add any other plugins here
  },
})
```

**Comment by dpetka2001** [comment]
The `minimal` style defines explicitly the highlight group it uses and it's the `NormalFloat` highlight group as you can see [here](https://github.com/folke/snacks.nvim/blob/98df370703b3c47a297988f3e55ce99628639590/lua/snacks/notifier.lua#L177-L178). If you change that, it will also change the background of the notification. 

So, it seems it is this way by design. If you comment out those 2 lines, then it will follow the default highlight groups. But it looks like Folke made this choice deliberately.

**Comment by dpetka2001** [comment]
Yeah, I don't know what to tell you. Better leave this open and let maintainer decide what the best approach is even though it doesn't seem to be a bug.

**Comment by gennaro-tedesco** [comment]
Notice that `style` accepts a function as well, see [here](https://github.com/folke/snacks.nvim/blob/d312053f78b4fb55523def179ac502438dd93193/lua/snacks/notifier.lua#L165). This means you could override it and define your own highlight group:
```lua
notifier = {
  style = function(buf, notif, ctx)
    ctx.opts.border = "none"
    local whl = ctx.opts.wo.winhighlight
    ctx.opts.wo.winhighlight = whl:gsub(ctx.hl.msg, "NormalFloat")
    vim.api.nvim_buf_set_lines(buf, 0, -1, false, vim.split(notif.msg, "\n"))
    vim.api.nvim_buf_set_extmark(buf, ctx.ns, 0, 0, {
      virt_text = { { notif.icon, ctx.hl.icon } },
      virt_text_pos = "right_align",
    })
  end,
  ..
}
```
where you can just copy and paste the above and amend `ctx.opts.wo.winhighlight = whl:gsub(ctx.hl.msg, "NormalFloat")` accordingly.
